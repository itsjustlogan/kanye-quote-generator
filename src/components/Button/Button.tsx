import React from 'react'

interface Props {
  label: string
  handleClick: () => void
}

export default function Button({ label, handleClick }: Props) {
  return (
    <div className="grid md:pt-40">
      <button
        data-testid="button"
        type="button"
        className="mx-auto border-none border-radius-medium bg-red-200 text-white font-bld py-3 px-2 font-lg xl:font-xl cursor-pointer hover:scale-90 transition-1"
        onClick={handleClick}
      >
        {label}
      </button>
    </div>
  )
}
