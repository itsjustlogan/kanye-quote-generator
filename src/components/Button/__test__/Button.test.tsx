/* eslint-disable comma-dangle */

import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import { vi } from 'vitest'
import Button from '..'

describe('Button component', () => {
  afterEach(() => {
    cleanup()
  })

  it('should render correctly', () => {
    expect(Button).toBeTruthy()

    const fn = vi.fn()

    const { getByTestId, asFragment, rerender } = render(
      <Button label="click me" handleClick={fn} />
    )
    expect(getByTestId('button')).toBeTruthy()
    expect(asFragment().textContent).toContain('click me')

    rerender(<Button label="click me again!" handleClick={fn} />)
    expect(asFragment().textContent).toContain('click me again!')

    rerender(<Button label="custom button" handleClick={fn} />)
    expect(asFragment().textContent).toContain('custom button')
  })

  it('should call function when clicked', () => {
    const fn = vi.fn()
    const { getByTestId } = render(<Button label="click me" handleClick={fn} />)
    fireEvent.click(getByTestId('button'))
    expect(fn).toHaveBeenCalledTimes(1)
    fireEvent.click(getByTestId('button'))
    fireEvent.click(getByTestId('button'))
    expect(fn).toHaveBeenCalledTimes(3)
  })

  it('should match snapshot', () => {
    const fn = vi.fn()
    const { asFragment } = render(<Button label="click me" handleClick={fn} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
