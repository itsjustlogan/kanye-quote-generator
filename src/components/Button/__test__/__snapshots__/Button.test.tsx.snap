// Vitest Snapshot v1

exports[`Button component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="grid md:pt-40"
  >
    <button
      class="mx-auto border-none border-radius-medium bg-red-200 text-white font-bld py-3 px-2 font-lg xl:font-xl cursor-pointer hover:scale-90 transition-1"
      data-testid="button"
      type="button"
    >
      click me
    </button>
  </div>
</DocumentFragment>
`;
